jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1420,7 +1432,9 @@ code_blockt gen_nondet_array_init(
     init_array_expr =
       typecast_exprt(init_array_expr, pointer_type(element_type));
 
-  if(element_type.id() == ID_pointer || element_type.id() == ID_c_bool)
+  if(
+    element_type.id() == ID_pointer || element_type.id() == ID_c_bool ||
+    (element_type == java_char_type() && string_printable))
```

Comment:
I think we shouldn't do that. Sorry, I should have raised that earlier...
But this re-introduces an initialisation loop on all character arrays (as I imagine `--string-printable` will be universally used).
This will affect general performance, and also constrains nondeterministic char arrays to be smaller than the chosen unwind value. After a brief scan over Tika, I'm not worried that we'll lose much coverage because of the latter (because methods that have a char array as input usually contain a loop to deal with the array anyway), but the former is more of a concern I think.

Do we really need this feature to work for arrays? We decided otherwise in https://github.com/diffblue/test-gen/blob/develop/doc/cl-options.md#--java-assume-inputs-interval-lu-or-l-or-u so I guess the answer here will be the same?

Reply:
If it had been harder to implement, I would have pushed back and said that maybe we should ask Peter if this really needs to work on arrays :) so in my opinion, if it will be much worse, then no, we don't need to do it.

Reply:
It's not clear what the performance handicap really is, it's more of a moral issue...
Can we accept to have loops in our society?
If you don't mind reverting it (and documenting the limitation), then it will be more consistent with `--java-assume-inputs-interval`.

Reply:
For `--java-assume-inputs-interval` we decided not to do it for arrays for now until we have a solution that works for arrays without re-introducing an initialisation loop,

Reply:
OK, reverted for now, but left the tests marked as FUTURE.

@allredj Can you suggest where to document it?

Reply:
Yes, I am the doc man.
I would exactly as it was done for `--java-assume-inputs-interval`: https://github.com/diffblue/test-gen/blob/develop/doc/cl-options.md#--java-assume-inputs-interval-lu-or-l-or-u
with a `Limitation` subsection (I personally think the Jira ticket is optional, but others might feel more strongly about it).

Reply:
OK, I've done it in test-gen, but I'll look at the limitation subsection thing. If you don't think it should be in CBMC, can you reapprove the PR given that I've removed the array stuff?

Reply:
No, there's no good place in cbmc to document that kind of thing.